telemetry_package/main.py

```diff
@@ -3,164 +3,203 @@
 
-        After building the UNIX socket, this function creates three subscription: one to /rosout, one to TopicNames.Control and one to TopicNames.Heartbeat.  
+        After building the UNIX socket, this function creates two subscription: /rosout and TopicNames.Heartbeat.
 
-        Args: 
+        Args:
             None
-        
-        Returns: 
+
+        Returns:
             None
-        '''
+        """
 
         super().__init__('telemetry_node')
-        qos_profile = QoSProfile( 
+
+        # Default QoS for the GroundStation - Node
+        qos_profile = QoSProfile(
             history=QoSHistoryPolicy.KEEP_LAST,
             depth=15,
             reliability=QoSReliabilityPolicy.BEST_EFFORT,
-            durability=QoSDurabilityPolicy.VOLATILE, 
-            liveliness = QoSLivelinessPolicy.AUTOMATIC
+            durability=QoSDurabilityPolicy.VOLATILE,
+            liveliness=QoSLivelinessPolicy.AUTOMATIC
         )
 
-        self.chosen_socket_path = ''
-
-        if sys.argv[1] == '': self.chosen_socket_path = DEFAULT_UNIX_SOCKET_PATH
-        else: self.chosen_socket_path = sys.argv[1]
+        # For tracking the timestamps between the last received messages for throttled log-message names
+        self._message_throttle_table = {}
 
+        # Assume second and only arg as explicit path to unix socket
+        if len(sys.argv) > 1:
+            self.chosen_socket_path = sys.argv[1]
+        else:
+            self.chosen_socket_path = DEFAULT_UNIX_SOCKET_PATH
 
-        if os.path.exists(self.chosen_socket_path): 
+        # Check if the socket already exists
+        if os.path.exists(self.chosen_socket_path):
             raise FileExistsError('Socket already exists.')
-            
-        self.server_sock = socket.socket(socket.AF_UNIX, socket.SOCK_STREAM)
 
         try:
+            # Create the socket on the given path and listen for one connection
+            self.server_sock = socket.socket(
+                socket.AF_UNIX, socket.SOCK_STREAM)
             self.server_sock.bind(self.chosen_socket_path)
+            print(
+                f"Listening for connection on the UNIX-socket at {self.chosen_socket_path} ...")
             self.server_sock.listen(1)
 
             # Accept the connection
             self.client_sock, _ = self.server_sock.accept()
-            print('Connection accepted. Creating subscription')
-
+            print('Connection accepted. Creating subscriptions...')
 
+            # Subscribe to /rosout
             self.rosout_subscription = self.create_subscription(
                 Log,
                 '/rosout',
-                self.__rosout_callback, 
-                qos_profile = qos_profile)
-            
+                self.__rosout_callback,
+                qos_profile=qos_profile)
+
             self.rosout_subscription
             print('Created subscription to "/rosout".')
 
-
-            self.control_subscription = self.create_subscription(

```

Comment:
Out of curiosity, why did you decide to leave the subscription to control out? If i remember correctly, you and @Philipp66904 said that a subscription to the Control topic would have been useful. I do not recall your reasoning behind that statement, though.

Reply:
Yes, I said that at the beginning.
But later I realized, that the `Control` topic is already kind of integrated in the `heartbeat` topic (in the `active` flag, to be precise), so we no longer need the `Control` topic in here.

Reply:
Thank you, that explains it